new-components/scanners/bandit/internal/transformer/transformer.go

```diff
@@ -0,0 +1,314 @@
+		LineRange       []int     `json:"line_range,omitempty"`
+		MoreInfo        string    `json:"more_info,omitempty"`
+		TestID          string    `json:"test_id,omitempty"`
+		TestName        string    `json:"test_name,omitempty"`
+	}
+
+	banditTransformer struct {
+		targetType     target.TargetType
+		clock          clockwork.Clock
+		rawOutFilePath string
+	}
+)
+
+// BanditTransformerWithClock allows customising the underlying clock.
+func BanditTransformerWithClock(clock clockwork.Clock) BanditTransformerOption {
+	return func(g *banditTransformer) error {
+		if clock == nil {
+			return errors.Errorf("invalid nil clock")
+		}
+		g.clock = clock
+		return nil
+	}
+}
+
+// GosecTransformerWithTarget allows customising the underlying target type.
+func BanditTransformerWithTarget(target target.TargetType) BanditTransformerOption {
+	return func(g *banditTransformer) error {
+		if target == "" {
+			return errors.Errorf("invalid empty target")
+		}
+		g.targetType = target
+		return nil
+	}
+}
+
+// GosecRawOutFilePath allows customising the underlying raw out file path.
+func BanditRawOutFilePath(path string) BanditTransformerOption {
+	return func(g *banditTransformer) error {
+		if path == "" {
+			return errors.Errorf("invalid raw out file path")
+		}
+		g.rawOutFilePath = path
+		return nil
+	}
+}
+
+// New returns a new bandit transformer.
+func New(opts ...BanditTransformerOption) (*banditTransformer, error) {
+	rawOutFilePath, err := env.GetOrDefault(
+		"BANDIT_RAW_OUT_FILE_PATH",
+		"bandit.json",
+		env.WithDefaultOnError(true),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	tt, err := env.GetOrDefault(
+		"BANDIT_TARGET_TYPE",
+		target.TargetType_Repository,
+		env.WithDefaultOnError(true),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	t := banditTransformer{
+		rawOutFilePath: rawOutFilePath,
+		targetType:     target.TargetType(tt),
+		clock:          clockwork.NewRealClock(),
+	}
+
+	for _, opt := range opts {
+		if err := opt(&t); err != nil {
+			return nil, errors.Errorf("failed to apply option: %w", err)
+		}
+	}
+
+	switch {
+	case t.rawOutFilePath == "":
+		return nil, errors.New("invalid empty raw output file")
+	case t.targetType == "":
+		return nil, errors.New("invalid empty target type")
+	}
+
+	return &t, nil
+}
+
+// Transform transforms raw sarif findings into ocsf vulnerability findings.
+func (b *banditTransformer) Transform(ctx context.Context) ([]*ocsf.VulnerabilityFinding, error) {
+	logger := component.LoggerFromContext(ctx)
+
+	logger.Debug("preparing to parse raw bandit output...")
+
+	inFile, err := os.ReadFile(b.rawOutFilePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, errors.Errorf("raw output file '%s' not found", b.rawOutFilePath)
+		}
+		return nil, errors.Errorf("failed to read raw output file '%s': %w", b.rawOutFilePath, err)
+	}
+
+	var results BanditOut
+	if err := json.Unmarshal(inFile, &results); err != nil {
+		log.Fatal(err)
```

Comment:
Can we avoid using these statements and return an error so that the SDK can handle it correctly?